src/seeds.history.cpp

```diff
@@ -113,202 +126,236 @@ void history::trxentry(name from, name to, asset quantity) {
-  }
 
-}
+void history::savepoints(uint64_t id, uint64_t timestamp) {
+  require_auth(get_self());
 
-// return true if anything was cleaned up, false otherwise
-bool history::clean_old_tx(name org, uint64_t chunksize) {
-  
-  auto now = eosio::current_time_point().sec_since_epoch();
-  auto cutoffdate = now - utils::moon_cycle * 3;
+  auto date = eosio::time_point_sec(timestamp / 86400 * 86400);
+  uint64_t day = date.utc_seconds;
 
-  org_tx_tables orgtx(get_self(), org.value);
-  uint64_t count = 0;
-  auto transactions_by_time = orgtx.get_index<"bytimestamp"_n>();
-  auto tx_time_itr = transactions_by_time.begin();
-  bool result = false;
-  while(tx_time_itr != transactions_by_time.end() && tx_time_itr->timestamp < cutoffdate && count < chunksize) {
-    tx_time_itr = transactions_by_time.erase(tx_time_itr);
-    result = true;
-    count++;
-  }  
-  return result;
-  
-}
+  daily_transactions_tables transactions(get_self(), day);
+  auto transactions_by_from_to = transactions.get_index<"byfromto"_n>();
 
-void history::orgtxpoints(name org) {
-  require_auth(get_self());
-  orgtxpt(org, 0, 200, 0);
-}
+  auto titr = transactions.find(id);
+  check(titr != transactions.end(), "transaction not found");
+  name from = titr -> from;
+  name to = titr -> to;
 
-void history::orgtxpt(name org, uint128_t start_val, uint64_t chunksize, uint64_t running_total) {
-  require_auth(get_self());
+  auto uitr_from = users.find(from.value);
+  auto uitr_to = users.find(to.value);
 
-  org_tx_tables orgtx(get_self(), org.value);
+  uint64_t max_number_transactions = config_get("htry.trx.max"_n);
+
+  uint128_t from_to_id = (uint128_t(titr -> from.value) << 64) + titr -> to.value;
   uint64_t count = 0;
 
-  if (start_val == 0 && clean_old_tx(org, chunksize)) {    // Step 0 - remove all old transactions
-    fire_orgtx_calc(org, 0, chunksize, running_total);
-    return;
+  auto ft_itr = transactions_by_from_to.find(from_to_id);
+  auto current_itr = ft_itr;
+
+  while (ft_itr != transactions_by_from_to.end() && 
+      count <= max_number_transactions && 
+      ft_itr -> from == from && ft_itr -> to == to) {
+
+    if (ft_itr -> volume < current_itr -> volume) {
+      current_itr = ft_itr;
+    }
+
+    ft_itr++;
+    count++;
   }
 
-  // Step 1 - add up values
-  uint64_t max_quantity = config_get("org.trx.max"_n);
-  uint64_t max_number_of_transactions = config_get("org.trx.div"_n);;
+  int64_t from_points = int64_t(titr -> from_points);
+  int64_t to_points = int64_t(titr -> to_points);
+  int64_t qualifying_volume = int64_t(titr -> qualifying_volume);
 
-  auto transactions_by_other = orgtx.get_index<"byother"_n>();
-  auto tx_other_itr = start_val == 0 ? transactions_by_other.begin() : transactions_by_other.lower_bound(start_val);
+  if (count > max_number_transactions) {
+    from_points -= current_itr -> from_points;
+    to_points -= current_itr -> to_points;
+    qualifying_volume -= current_itr -> qualifying_volume;
```

Comment:
So this could either end up in some positive values, or it could be all 0 if the new transaction is the smallest transaction

Then it deletes the smallest, so ends back up with 2 transactions

Reply:
Yes, that's correct. But it also could be negative values, those are calculated in order to adjust the total points, so even if they become negative, the total transaction points calculated will never be negative. I realized that a transaction could have a grater volume, but not necessarily give more transaction points, I thought we had to keep the 2 transactions with the biggest volume, is that right?

Reply:
Yes on the 2tx

I'll trust that it works on the others, not sure how they can be negative ;)